### ISSUE TITLE
Devils Sight and Darkvision create an infinite Darkvision vision range (but no actual token visibility to that range)

### ISSUE THREAD

**Issue by dineshm72** [dev7355608/vision-5e#111]
Using FTC with only Vision 5e and libraries active.  I have a character who has both 60' of Darkvision (from elf) and 120' of Devil's Sight (from warlock invocation).

Characters with only Darkvision correctly get 60' of Darkvision detection by default:
![image](https://github.com/user-attachments/assets/67b099f4-c10f-45a8-8997-c74360dca010)

For some reason, this combination of Darkvision and DS is creating a default detection mode of infinite darkvision in addition to the 120' of Devil's Sight detection:
![image](https://github.com/user-attachments/assets/aed533ff-a335-4a87-9a70-828089d04172)

(As an aside, DDB incorrectly shows this as 120 feet of darkvision on the char sheet, but I don't think that matters at all.)
![image](https://github.com/user-attachments/assets/703cb138-e674-4c0c-ac75-b0e725282103)

The visuals being generated seem to be mostly right.  In this example, the token on the right is the one with both DV and DS.  The two tokens on the left are 120 and 125 feet away.  The wall behind them is 140 feet away.  GM view with nothing selected.
![image](https://github.com/user-attachments/assets/0f5c91de-38d1-4f95-8622-7530b5c8fad9)

When switched to Devil's Sight Vision mode for the right hand token, it seems to work perfectly, token config shows the Vision range at 120', colored vision out to 120', one token visible, one not visible and not detected by anything.  
![image](https://github.com/user-attachments/assets/416a7d50-717a-436e-86d6-a42b92546dc6)

When set to Darkvision, the token config indicates an infinite vision range (which is wrong for the char sheet, but correct to the detection mode setting).  The dim light polygon is generated out to infinity (or at least to 140 feet), which is wrong to the character sheet but correct to the config dialog.  But even weirder, tokens in that dim light are actually only visible out to 120 feet, past that there is a dim light polygon being generated apparently, but no tokens past 120' in that polygon are actually getting illuminated by it (i.e. the left most token in this pic is at 125 feet and standing in the darkvision dim light polygon and so should be visible to the token on the right, but isn't).  But detection (correctly, I think) only works out to 120' (presumably from the DS).  
![image](https://github.com/user-attachments/assets/446166b8-e0ed-40f0-ba28-96fb2ce0a76e)

Manually overriding the Darkvision Detection mode to 60' seems to give what I'd expect, 60 feet of vision range, 60 feet of dim light polygon, nothing illuminated by dim light beyond that, then a token detected (by the Devil's Sight) out at 120, and no detection of the token at 125 feet.
![image](https://github.com/user-attachments/assets/b4d0b036-36ae-4e03-ae43-ded7926bf8e7)

So, to summarize, I think there are 2 issues.  One, this char with 60' DV and 120' DS is getting generated with an infinite darkvision range.  Two, tokens at more than 120' with the incorrect defaults should be visible, but aren't for some reason.

**Comment by dineshm72** [comment]
I should add that the notes say:
Note: Remove the active effect that gives 120 feet Darkvision from the Invocation: Devil's Sight feat if it exists: Devil's Sight doesn't increase the range of Darkvision.

But my invocation only has the CPR AE listed (and CPR isn't even enabled at the time of testing):
![image](https://github.com/user-attachments/assets/bcb674e7-ac37-4b04-b55d-d6876453cdf4)

**Comment by dineshm72** [comment]
Finally, just to confirm: the proto token detection modes table is supposed to be blank, correct?  And then gets generated automatically when the token is dropped?  That seems to be what I am seeing.

Adding in the override for DV in that blank table does seem to get saved and does overnight all new tokens, which is perfect.  I was just a little surprised to find it blank.

**Comment by dev7355608** [comment]
On token creation the prototype data is cloned. So any detection modes the prototype token had are copied to the token, which then override the detection mode ranges that automatically calculated from the actor by Vision 5e. So with Vision 5e you actually don't want any entries in the prototype's detection modes.

**Comment by dev7355608** [comment]
You say the prototype token that infinite Darkvision configured? But the fact that the Darkvision entry in the token config is disabled tells me it can't be from the prototype token: only automatically calculated range from the actor are disabled. So that's odd. Not sure how you can end up with infinite darkvision. What does `_token.document.actor.system.attributes.senses` return?

**Comment by dineshm72** [comment]
No, in the prototype token the detection modes table is blank/empty.  When the token gets dropped, it generates the detection modes shown above, with the infinite darkvision.

I then manually put a 60' darkvision detection on the token prototype, and it behaved as you described and as I hoped - it overrode the default darkvision infinite with my manual 60' darkvision.  So I at least have a way to manually fix it for this PC actor.

If I then later deleted that manual entry, it reverted back to the infinite darkvision.

The console command above returns: 
![image](https://github.com/user-attachments/assets/801ea156-9106-49d2-ab22-b703352e8f18)

**Comment by dev7355608** [comment]
Is `darkvision` undefined even if you disable all modules? If that that would be a system bug, I think. The sense ranges should always be a number. What version of dnd5e do you use?

**Comment by dineshm72** [comment]
yep (at least with everything off but FTC):

![image](https://github.com/user-attachments/assets/aa61b6b1-5165-46f0-8a59-b8e66e0366e8)

On dnd 3.3.1

**Comment by dev7355608** [comment]
Since 3.2.0 the prepared sense ranges should always be numeric. Are you sure there isn't an active effect that somehow breaks the darkvision range? Look at `_token.document.actor.overrides` to be sure.

**Comment by dineshm72** [comment]
I'm not exactly sure what I'm looking for here, but here is what that command returns:

![image](https://github.com/user-attachments/assets/83a2af09-3a23-4226-9fa2-b1463d3974ec)

The CPR and midi flags are expected I think.  There is an effect for darkvision: undefined but it is a system one, I have no idea why it is there.  

As info, here is the effects tab:
![image](https://github.com/user-attachments/assets/7701e836-180d-44f4-8353-248645aabb5c)

**Comment by dev7355608** [comment]
One of these active effects must affect darkvision. Doesn't the Belt of Dwarvenkind give darkvision?

**Comment by dineshm72** [comment]
D'oh!  You are of course correct, it seems that having that item attuned and equipped is causing the DV setting to go undefined.

I tried deleting and bringing in a fresh copy and it's still doing it, so I'm guessing maybe it's an issue with the item or the DDBI import.  I'll have a loot and then take it to MrPrimate.

Sorry to bother you with this!

**Comment by dev7355608** [comment]
No problem!
